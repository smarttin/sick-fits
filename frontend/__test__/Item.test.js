import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ItemComponent from '../components/Item';

const fakeItem = {
  id: 'qwswedr',
  title: 'what i dread',
  price: 4000,
  description: 'testing is what i have to do so ...',
  image: 'test.jpg',
  largeImage: 'largeTestImage.jpg',
};

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  // it('renders the image properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const img = wrapper.find('img');
  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title);
  // });

  // it('renders and displays properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const priceTag = wrapper.find('ForwardRef(PriceTag)');
  //   expect(priceTag.children().text()).toBe('$40');
  //   expect(wrapper.find('ForwardRef(Title) a').text()).toBe(fakeItem.title);
  // });

  // it('renders out the buttons properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const buttonList = wrapper.find('.buttonList');
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link')).toHaveLength(1);
  //   expect(buttonList.find('AddToCart').exists()).toBe(true);
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true);
  //   // console.log(buttonList.debug());
  // });
});
