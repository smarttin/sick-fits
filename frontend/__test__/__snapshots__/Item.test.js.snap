// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ForwardRef(ItemStyles__Item)>
  <img
    alt="what i dread"
    src="test.jpg"
  />
  <ForwardRef(Title)>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "qwswedr",
          },
        }
      }
    >
      <a>
        what i dread
      </a>
    </Link>
  </ForwardRef(Title)>
  <ForwardRef(PriceTag)>
    $40
  </ForwardRef(PriceTag)>
  <p>
    testing is what i have to do so ...
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "qwswedr",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="qwswedr"
    />
    <DeleteItem
      id="qwswedr"
    >
      Delete this item
    </DeleteItem>
  </div>
</ForwardRef(ItemStyles__Item)>
`;
